### DISCUSSION TITLE
ESP 8266 Servo.h: Attaching the servo at a given angle

### DISCUSSION THREAD

**Discussion by AllanTee** [esp8266/Arduino#8699] [Q&A]
Does the Servo library support detaching and following attaching the servo motor again at exact desired angle without unwanted jumps of the motor? How to achieve this behavior?

**Comment by earlephilhower** [comment]
That's not possible with a servo.  It needs to be fed a continuous pulse stream to hold any position.

Why are you trying to disconnect it?  Just set a servo angle, do whatever you what, then some time later set a new angle.  No disconnects needed.

**Reply by AllanTee** [reply]
Hi Earle,
A servo motor, as I have learned, in a stationary state, makes too much noise (and draws more current) even when applied with a small amount of force. To avoid the noise, I have to detach the servo. Now even if the servo has been displaced by some external force in the meantime, I would like to continue driving it from the position it is currently in, without any sudden random movements (which are making unwanted noise).

To know the position angle of the servo, I soldered a fourth wire. Now I can measure the feedback value and calculate corresponding angle (or milliseconds). I can detach and attach it again. But in most cases it fails to attach the servo without movement. I have had the best results with the ESP 8266 and given library. I have also tried ESP32Servo and ServoEasing library with ESP32 but I am not getting the desired result with them. (Besides, it's overcomplicated).

I understand that such a use case is not quite common and the developers have not provided corresponding methods for this. But, I have also read that people have asked similar questions on the forums. It would be great if such an opportunity opened up.

**Reply by earlephilhower** [reply]
That sounds odd, but I suppose if there is no torque on the servo and you drop power it may just sit where you had it.

It's been a while, but looking at the Servo sources https://github.com/esp8266/Arduino/blob/d3eddeb5014d04aa9e61b97474ef98d0214eab1a/libraries/Servo/src/Servo.cpp#L89-L:99

You could probably do a `stopWaveform` on the pin which will disable the pulses on the pin.

**Reply by AllanTee** [reply]
Hi Earle,

After checking the library, I found that there is another method that I haven't seen an example of yet:
_uint8_t Servo::attach(int pin, uint16_t minUs, uint16_t maxUs, int **value**)_.
The fourth argument was exactly what I was looking for. Now I can enter the angle of my servo position and that's it. It meets my needs one hundred percent. 
Thank you very much!
